apps/condo/domains/common/components/HeaderActions.tsx

```diff
@@ -16,16 +17,25 @@ export const TabsAuthAction: React.FC<ITabsActionsProps> = (props) => {
     const registerTab = intl.formatMessage({ id: 'pages.auth.RegistrationTitle' })
     const signInTab = intl.formatMessage({ id: 'pages.auth.SignInTitle' })
 
+    const router = useRouter()
+    const { query: { next }  } = router
+    const isValidNextUrl = next && !Array.isArray(next) && isSafeUrl(next)
+
     return (
         <RemoveTabsLineWrapper>
             <Tabs
                 defaultActiveKey={currentActiveKey}
-                onChange={(activeKey) => Router.push(activeKey)}
+                onChange={(activeKey) => {
+                    if (activeKey === 'signin') {
+                        router.push(isValidNextUrl ? `/auth/signin?next=${encodeURIComponent(next)}` : '/auth/signin')
+                    } else if (activeKey === 'register') {
+                        router.push(isValidNextUrl ? `/auth/register?step=inputPhone&next=${encodeURIComponent(next)}` : '/auth/register')
```

Comment:
Should you add the "step" parameter here?

```suggestion
                        router.push(isValidNextUrl ? `/auth/register?step=inputPhone&next=${encodeURIComponent(next)}` : '/auth/register?step=inputPhone')
```